presto-native-execution/presto_cpp/main/common/Configs.cpp

```diff
@@ -648,7 +651,14 @@ std::string NodeConfig::nodeEnvironment() const {
 }
 
 std::string NodeConfig::nodeId() const {
-  return requiredProperty(kNodeId);
+  auto resultOpt = optionalProperty(kNodeId);
```

Comment:
Nit : Can we have just one static variable that is assigned from optionalProperty(kNodeId) or a generated value (instead of separate resultOpt and nodeId) ?

Reply:
My thought is that we can skip the static variable if the user specifies the node.id value. There is no benefit to adding a static variable here.
For a generated value, we have to use a static and cannot avoid it.

Reply:
Ah I see... Though I feel the following code is a bit more readable.

```
auto resultOpt = optionalProperty(kNodeId);
if (resultOpt.hasValue()) {
  return resultOpt.value();
}

// nodeId must be singleton.
static auto nodeId =
   boost::lexical_cast<std::string>(boost::uuids::random_generator()());
 return nodeId;
```

Reply:
I made this refactor.